docs/pages/access-controls/reference.mdx

```diff
@@ -357,9 +357,41 @@ allow:
     - resources:
         - token
       verbs: [list, create, read, update, delete]
+```
+
+### Allowing access to token resources
 
+If you configure a role that allows tokens to be created, users assigned to the 
+role can create tokens to provision any type of Teleport resource. 
+For example, you might create a role with the following configuration to enable assigned
+users to enroll servers:
+
+```yaml
+kind: role
+version: v7
+metadata:
+  name: enroll-servers
+spec:
+  allow:
+    node_labels:
+      'env': 'us-lab'
+    rules:
+      - resources: [token]
+        verbs: [list, create, read, update, delete]
+  deny: {}
 ```
 
+With these permissions, users assigned to the role can generate tokens to enroll  
+a server, application, or database, establish a trust relationship between a root 
+cluster and a new Teleport Proxy Service or leaf cluster, or configure a device 
+trust relationship for a managed device.
```

Comment:
Have we verified the device trust part? Generating device enroll tokens shoudl require the `device:enroll` permission, so I don't believe that `token:create` would be enough.

@codingllama ?

Reply:
Zac is correct, we do require the device/create_enroll_token permission to enroll new devices, good old "token" won't do. The device_admin role is an example of that: https://goteleport.com/docs/access-controls/device-trust/device-management/#dedicated-device-admin-role.

Reply:
@zmb3 @codingllama Just a quick followup: this caveat was for configuring a device trust relationship in the original doc request. Is that the same as enrolling a device?

Reply:
Enrollment is what turns a registered device (a device that Teleport knows about) into a trusted device (a device that passes "trusted device" checks). The "token" resource has no influence in the device trust subsystem, so I think the comment stands.